src/display/editor/stamp.js

```diff
@@ -70,20 +70,30 @@ class StampEditor extends AnnotationEditor {
 
   #getBitmap() {
     if (this.#bitmapId) {
-      this._uiManager.imageManager.getFromId(this.#bitmapId).then(data => {
-        if (!data) {
-          this.remove();
-          return;
-        }
-        this.#bitmap = data.bitmap;
-        this.#createCanvas();
-      });
+      this._uiManager.enableWaiting(true);
+      this._uiManager.imageManager
+        .getFromId(this.#bitmapId)
+        .then(data => {
+          if (!data) {
+            this.remove();
+            return;
+          }
+          this.#bitmap = data.bitmap;
+          this.#createCanvas();
+        })
+        .finally(() => {
+          this._uiManager.enableWaiting(false);
+          if (this.#canvas) {
+            this.div.focus();
+          }
+        });
```

Comment:
How about adding a new helper e.g.
```javascript
#getBitmapDone() {
  this._uiManager.enableWaiting(false);
  if (this.#canvas) {
    this.div.focus();
  }
}
```
since then all of these new finally-blocks could be replaced with something like
```suggestion
        .finally(() => this.#getBitmapDone());
```